File: python/command_injection.py
import os
import subprocess

cmd = input('Enter command to run: ')
# CWE-78: user input passed directly to os.system without validation.
cmd_parts = cmd.split()
ALLOWED_COMMANDS = {'ls', 'pwd', 'echo'}
if not cmd_parts or cmd_parts[0] not in ALLOWED_COMMANDS:
    print("Error: Command not allowed")
    exit(1)
subprocess.run(cmd_parts, shell=False, check=False)

File: javascript/command_injection.js
const { exec } = require('child_process');
const readline = require('readline');

const rl = readline.createInterface({ input: process.stdin, output: process.stdout });
rl.question('Enter command to run: ', (answer) => {
  // CWE-78: user input passed directly to exec without validation.
  exec(answer, (err, stdout, stderr) => {
    if (err) { console.error('error:', err); return; }
    process.stdout.write(stdout);
  });
  rl.close();
});
